mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java

```diff
@@ -731,21 +731,24 @@ private MoveResult move(Iterator<MailboxMessage> originalRows, MailboxSession se
     private SortedMap<MessageUid, MessageMetaData> copy(MessageRange set, StoreMessageManager to, MailboxSession session) throws MailboxException {
         IteratorWrapper<MailboxMessage> originalRows = new IteratorWrapper<>(retrieveOriginalRows(set, session));
 
-        SortedMap<MessageUid, MessageMetaData> copiedUids = collectMetadata(to.copy(originalRows, session));
+        if (originalRows.hasNext()) {
```

Comment:
I don't agree with this if.
All the code works great when originalRows is empty but the event broadcasting, we should just prevent firing the events on Noop, WDYT ?